spec/lib/dsc_utils.rb

```diff
@@ -43,75 +43,75 @@ def get_dsc_vendor_resource_abs_path(dsc_module, expect_ps_failure)
   dsc_prop_merge = '@{'
   dsc_properties.each do |k, v|
     dsc_prop_merge << "\"#{k}\"="
 
-    if v =~ /^\$/
-      dsc_prop_merge << "#{v};"
-    elsif v =~ /^@/
-      dsc_prop_merge << "#{v};"
-    elsif v =~ /^(-|\[)?\d+$/
-      dsc_prop_merge << "#{v};"
-    elsif v =~ /^\[.+\].+$/     #This is for wacky type conversions
-      dsc_prop_merge << "#{v};"
-    else
-      dsc_prop_merge << "\"#{v}\";"
-    end
+    dsc_prop_merge << if v =~ %r{^\$}
+                        "#{v};"
+                      elsif v =~ %r{^@}
+                        "#{v};"
+                      elsif v =~ %r{^(-|\[)?\d+$}
+                        "#{v};"
+                      elsif v =~ %r{^\[.+\].+$?} # This is for wacky type conversions
+                        "#{v};"
+                      else
+                        "\"#{v}\";"
+                      end
   end
 
   dsc_prop_merge.chop!
   dsc_prop_merge << '}'
 
-  #Compose strings for PS command.
+  # Compose strings for PS command.
   dsc_command = "Invoke-DscResource -Name #{dsc_resource_type} " \
                 "-Method #{dsc_method} " \
                 "-ModuleName #{dsc_module} " \
-                "-Verbose " \
+                '-Verbose ' \
                 "-Property #{dsc_prop_merge}"
 
-  return "if (#{dsc_command}) { exit 0 } else { exit 1 }"
+  "if (#{dsc_command}) { exit 0 } else { exit 1 }"
 end
 
 def _exec_dsc_script(ps_script, expect_ps_failure, &block)
-  #Init
+  # Init
   temp_script = 'temp.ps1'
   utf8_ps_script = "\xEF\xBB\xBF".force_encoding('UTF-8') + ps_script.force_encoding('UTF-8')
   ps_launch = "& C:/temp/#{temp_script}"
 
-  #Create remote file with UTF-8 BOM
-  create_windows_file("C:/temp", temp_script, utf8_ps_script)
+  # Create remote file with UTF-8 BOM
+  create_windows_file('C:/temp', temp_script, utf8_ps_script)
 
-  #Execute PowerShell script on host
-  result = run_shell("powershell.exe -NoProfile -NoLogo -Command \"#{ps_launch}\"", :expect_failures => expect_ps_failure)
+  # Execute PowerShell script on host
+  result = run_shell("powershell.exe -NoProfile -NoLogo -Command \"#{ps_launch}\"", expect_failures: expect_ps_failure)
 
-  #Also, let additional checking be performed by the caller.
+  # Also, let additional checking be performed by the caller.
   if block_given?
     case block.arity
-      #block with arity of 0, just hand back yourself
-      when 0
-        yield self
-      #block with arity of 1 or greater, hand back the result object
-      else
-        yield result
+      # block with arity of 0, just hand back yourself
+    when 0
+      yield self
+      # block with arity of 1 or greater, hand back the result object
+    else
+      yield result
     end
   end
   result
 end
 
-def assert_dsc_resource(dsc_resource_type, dsc_module, expect_ps_failure = false, expect_dsc_failure = false, dsc_properties)
+def assert_dsc_resource(dsc_resource_type, dsc_module, dsc_properties, expect_ps_failure = false, expect_dsc_failure = false)
```

Comment:
the ordering of this has changed, is it intentional

Reply:
it looks like it was changed in other places too, so i'll assume it is intentional

Reply:
I remember the linting failing if I didn't move this around, but it's distant enough I don't directly remember exactly now. :(